modules/ROOT/pages/rtm-agent-app-validator.adoc

```diff
@@ -0,0 +1,616 @@
+
+    // Validator configuration values
+    String business = (String) args.get("business");
+    String welcomeMessage = (String) args.get("welcomeMessage");
+    String errorMessage = (String) args.get("errorMessage");
+
+    LOGGER.info("Values injected by the service:");
+    LOGGER.info("\tApplication name: {}", applicationName);
+    LOGGER.info("\tApplication file path: {}", applicationFilePath);
+    LOGGER.info("\tApplication properties: {}", applicationProperties);
+
+    LOGGER.info("Validator configurations:");
+    LOGGER.info("\tBusiness: {}", business);
+    LOGGER.info("\tWelcome message: {}", welcomeMessage);
+    LOGGER.info("\tError message: {}", errorMessage);
+}
+----
+
+==== Update the mule-agent.yml File
+
+Add the validation configuration in the service:
+
+[source,java,linenums]
+----
+services:
+  mule.agent.application.validator.service:
+    enabled: true
+    validators:
+    - type: helloValidator
+      name: defaultHelloValidator
+      enabled: true
+      args:
+        business: finance
+        welcomeMessage: 'Hi, welcome to Finance Cluster Runtimes'
+        errorMessage: 'I am sorry, you are not allowed to be deployed in this Cluster'
+----
+
+=== Start the Mule
+
+The following logs show in the console:
+
+[source,java,linenums]
+----
+AgentArtifactValidatorService: Total validators loaded: 1
+AgentArtifactValidatorService: Validators listed by type:
+AgentArtifactValidatorService: 	Type: helloValidator Names: [defaultHelloValidator]
+----
+
+== Deploy an Application
+
+Now that the validator is ready, let’s deploy an application to see how the Agent Validator Service invokes the custom validator.
+
+Curl command:
+
+[source,console,linenums]
+----
+curl -X PUT 'http://localhost:9999/mule/applications/app-01' \
+-H 'Content-Type: application/json' \
+-d '{
+  "url": "file:/tmp/app-01.jar",
+  "configuration": {
+    "mule.agent.application.properties.service": {
+      "applicationName": "app-01",
+      "properties": {
+          "business": "finance"
+      }
+    }
+  }
+}'
+----
+
+Response:
+
+[source,console,linenums]
+----
+{
+  "application": {
+    "name": "app-01"
+  },
+  "status": "Deployment attempt started"
+}
+----
+
+Console logs:
+
+[source,console,linenums]
+----
+MuleAgentHelloValidator: Values injected by the service:
+MuleAgentHelloValidator:   Application name: app-01
+MuleAgentHelloValidator:   Application file path: /tmp/mule-received-artifact-1793122713159733968/app-01.jar
+MuleAgentHelloValidator:   Application properties: {business=finance}
+MuleAgentHelloValidator: Validator configurations:
+MuleAgentHelloValidator:   Allowed Business: finance
+MuleAgentHelloValidator:   Welcome message: Hi, welcome to Finance Cluster Runtimes
+MuleAgentHelloValidator:   Error message: I am sorry you not allowed to be deployed in this Cluster
+MuleAgentHelloValidator: Hi, welcome to Finance Cluster Runtimes
+----
+
+
+== Encrypt Values in the validators configuration
```

Comment:
Should this be validator configuration instead of validators configuration?

Reply:
Agree! Applied, and capitalized.